src/components/colorscale/color_attributes.js

```diff
@@ -64,14 +69,16 @@ module.exports = function makeColorScaleAttributes(context) {
         autocolorscale: extendDeep({}, colorScaleAttributes.autocolorscale, {
```

Comment:
Correct. It is a little more subtle at the moment. See this [line](https://github.com/plotly/plotly.js/blob/32293e36c7e4ee47afd9c033f7dc226ccc8d666e/src/components/colorscale/defaults.js#L37) in the colorscale defaults.

If one of `cmin` or `cmax` isn't valid, than `autoscale` is defaulted to `false`. But, if the users set `autoscale` to true, the missing `cmin` or `cmax` gets auto-calc'ed.